src/sentry/integrations/discord/client.py

```diff
@@ -45,12 +63,42 @@ def __init__(self):
     def prepare_auth_header(self) -> dict[str, str]:
         return {"Authorization": f"Bot {self.bot_token}"}
 
-    def overwrite_application_commands(self, commands: list[object]) -> None:
-        self.put(
-            APPLICATION_COMMANDS_URL.format(application_id=self.application_id),
-            headers=self.prepare_auth_header(),
-            data=commands,
-        )
+    def set_application_commands(self) -> None:
+        try:
```

Comment:
Usually in a client we want to only have the  request (`self.post(...)`) and have the try/catch/logging stuff happen wherever the client is called `integration.py` in this case).

Reply:
This lets you have more specific logging, logic, etc. if more than one places need to call the same endpoint

Reply:
Ah ok, thanks! Let me fix that...